### ISSUE TITLE
Browsers requests default browser setting on every launch

### ISSUE THREAD

**Issue by jonnyborbs** [Browsers-software/browsers#76]
**Describe the bug**
I have the following setup on my machine:
- System default browser is [Finicky](https://github.com/johnste/finicky) 
- There are about 20 URL patterns specified in the Finicky config that should always open in a specific browser
- The default browser in Finicky is set to Browsers

The net impact here is that those 20 URLs always bypass Browsers and open in Chrome, and anything else I click will open Browsers and prompt me where to go.

The problem is that since the system default is set to Finicky, Browsers prompts me on every URL click to set it as the default.

**Expected behavior**
Once I've told Browsers not to set itself as default once, it should not continue to ask

**System:**
 - OS: macOS Ventura 13.4.1(c)

**Comment by jonnyborbs** [comment]
I see there's a `rules` section in the config.json file but it's not clear to me what that does or how I'd use it: does that functionality replace Finicky in my scenario? Could I build regex patterns that automatically direct certain URLs to certain browsers here? If so, I could probably streamline my setup and also work around this issue at the same time.

**Comment by liias** [comment]
I will eventually plan to add a GUI for settings, but yep you found the settings file can do some things already.
There is _some_ documentation at https://github.com/Browsers-software/browsers/pull/34

I didn't pick regex for the syntax, but kind of glob-like patterns. I would be interested to get some feedback if that covers the need for you or not.

The easiest way to find the `profile` id is from the cache file. Or if you re-order the browser, then the id is also visible in the config file.

**Comment by jonnyborbs** [comment]
I'll look into that! I could share a [sanitized] version of my finicky.js if you think that would help you as a more experienced reader say if the rules functionality replicates what I'm doing.

In the meantime, is there a way to block Browsers from continuing to prompt to be set as default? That also handles the issue perfectly.

**Comment by jonnyborbs** [comment]
OK, I took some time today and converted all my regular expression browser rules to your glob format. It worked great for my basic use cases, which is just intercepting a bunch of root and sub domains and directing them to my "work" browser config. I could see situations where the regular expressions would be more powerful if you wanted more fine-grained control over these patterns (e.g. "open this Google doc in Chrome and this one in Safari) but those are probably edge cases.

A quick proper doc for this would be super helpful, even though it's just JSON and pretty clear from your PR.  That PR comment is about 90% of the way to being product-ready doc IMO, though it has a few minor errors that stop it from being used directly (missing commas, comments disallowed, some more examples)

As a result I've changed my default browser to Browsers and pulled Finicky out of the mix entirely. I still think better control over the default browser prompt would be worth adding to Browsers but it is no longer a specific problem I have.

Now, we just need to figure out how to get the menu to [exit more easily](https://github.com/Browsers-software/browsers/issues/77) when it loses focus and this is a perfect solution!
